mart/attack/adversary.py

```diff
@@ -150,35 +166,32 @@ def configure_gradient_clipping(
                 self.gradient_modifier(group["params"])
 
     @silent()
-    def forward(self, *, model=None, **batch):
-        batch["model"] = model
+    def forward(self, *, batch: torch.Tensor | list | dict, model: Callable):
```

Comment:
Is is possible to use `forward(self, *args, *, model: Callable, **kwargs):` instead? That better follows `LightningModule.forward`'s definition.

Reply:
Could you clarify why we want to follow `LightningModule.forward()` as we intend to run `training_step()` or `attack_step()`?

Reply:
Because I think it's better to be compatible with the expectation of how other things call `forward` now instead of changing the semantics. My suspicion is that others will want to use this adversary class and it is easy to understand if it just follows how others use `forward`.

Reply:
I guess this is fine if Adversary remains an internal/private class.

Reply:
Yes, I think we should keep it in this way. There are so many ways to implement `model.forward()` that we should not bear the burden of inferring the implementation here.